docs/src/main/sphinx/security/jwt.rst

```diff
@@ -0,0 +1,144 @@
+==================
+JWT authentication
+==================
+
+Trino can be configured to authenticate client access using `JSON web tokens
+<https://en.wikipedia.org/wiki/JSON_Web_Token>`_. A JWT is a small, web-safe
+JSON file that contains cryptographic information similar to a certificate,
+including:
+
+-  Subject
+-  Valid time period
+-  Signature
+
+Sites such as `jwt.io <jwt.io>`_ are available to help you decode and verify a
+JWT.
+
+A JWT is designed to be passed between servers as proof of prior authentication
+in a workflow like the following:
+
+* An end user logs into a client application and requests access to a server.
+* The server sends the user's credentials to a separate authentication service
+  that:
+
+  * validates the user
+  * generates a JWT as proof of validation
+  * returns the JWT to the requesting server
+
+* The same JWT can then be forwarded to other services to maintain the user's
+  validation without further credentials.
+
+.. important::
+
+    If you are trying to configure OAuth2 or OIDC, there is a dedicated system
+    for that in Trino, as described in :doc:`/security/oauth2`. When using
+    OAuth2 authentication, you do not need to configure JWT authentication,
+    because JWTs are handled automatically by the OAuth2 code.
+
+    A typical use for JWT authentication is to support administrators at large
+    sites who are writing their own single sign-on or proxy system to stand
+    between users and the Trino coordinator, where their new system submits
+    queries on behalf of users.
+
+Using JWT authentication
+------------------------
+
+Trino supports Base64 encoded JWTs, but not encrypted JWTs.
+
+There are two ways to get the encryption key necessary to validate the JWT
+signature:
+
+- Load the key from a JSON web key set (JWKS) endpoint service (the
+  typical case)
+- Load the key from the local file system on the Trino coordinator
+
+A JWKS endpoint is a read-only service that contains public key information in
+`JWK <https://datatracker.ietf.org/doc/html/rfc7517>`_ format. These public
+keys are the counterpart of the private keys embedded in JSON web tokens.
```

Comment:
There are no keys embedded in a JTW.  The JWT can be signed using a private key, and in that case we need the related public key to verify the signature.

Reply:
OK